### PR TITLE
Xamarin push

### PR DESCRIPTION
**Pull request by secondsun**
Adding in REALLLLY Basic GCM support.

This will need a lot of TLC because I know jack and crap about C#.  Please advice license headers, formatting, best practices, etc.

@edewit :eyes: ?

### CODE REVIEW

**Comment by edewit** [comment]
:eyes:

**Comment by edewit** [comment]
@secondsun should we keep the PushTest app?

**Comment by edewit** [comment]
created a pr with some formatting and changed names from java to c#

File: xamarin/FHXamarinAndroidSDK/Services/Device/DeviceService.cs [hunk 1]
```diff
@@ -96,6 +98,56 @@ public string GetPackageDir()
...
+		public PushConfig ReadPushConfig ()
+		{
+			Properties appProps = new Properties();
+			Stream input = null;
+			ILogService logger = ServiceFinder.Resolve<ILogService>();
+			try {
+				input = Application.Context.Assets.Open(APP_PROP_FILE);
+				appProps.Load(input);
+
+				string pushServerUrl = appProps.GetProperty("PUSH_SERVER_URL");
```
Comment (matzew) [inline xamarin/FHXamarinAndroidSDK/Services/Device/DeviceService.cs]: See RHMAP-1007 - there is no longer PUSH_SERVER_URL

that is interenally, in the SDK, created using the host property

Comment (matzew) [inline xamarin/FHXamarinAndroidSDK/Services/Device/DeviceService.cs]: @secondsun see https://github.com/feedhenry/fh-android-sdk/pull/70/files for the Android SDK change from @danielpassos

Comment (edewit) [inline xamarin/FHXamarinAndroidSDK/Services/Device/DeviceService.cs]: @matzew fixed that in my PR to this PR https://github.com/secondsun/fh-dotnet-sdk/pull/1

**Comment by edewit** [comment]
@secondsun could you rebase and merge this?

**Pull request merged**